pkgs/by-name/ms/msbuild/package.nix

```diff
@@ -83,7 +83,7 @@ mkPackage rec {
     # The provided libhostfxr.dylib is for x86_64-darwin, so we remove it
     rm artifacts/mono-msbuild/SdkResolvers/Microsoft.DotNet.MSBuildSdkResolver/libhostfxr.dylib
 
-    ln -s $(find ${dotnet-sdk.unwrapped}/share/dotnet -name libhostfxr${sharedLibrary}) artifacts/mono-msbuild/SdkResolvers/Microsoft.DotNet.MSBuildSdkResolver/
+    ln -s $(find ${dotnet-sdk.unwrapped}/share/dotnet/host -name libhostfxr${sharedLibrary}) artifacts/mono-msbuild/SdkResolvers/Microsoft.DotNet.MSBuildSdkResolver/
```

Comment:
This isn't anything to do with dotnet 8 specifically.  It's because the source-built sdk has two copies of this file.  The other one is in /packs.